src/Infrastructure/Stize.Infrastructure.Azure/Sql/SqlDatabaseExtensions.cs

```diff
@@ -67,36 +66,140 @@ public static SqlDatabaseBuilder ElasticPoolId(this SqlDatabaseBuilder builder,
+            Input<string> pointInTime = null)
+        {
+            /* Same comments SetAsRestore() method. 
+             * This method seems redundant as you don't need to set SourceDatabaseDeletionDate to restore a deleted database - the RestorableDroppedDatabaseId will suffice.
+             * "If sourceDatabaseId is the database�s original resource ID, then sourceDatabaseDeletionDate must be specified." - https://www.pulumi.com/docs/reference/pkg/azure-native/sql/database/#createmode_csharp
+             * ^ documentation states to assign SourceDatabaseId with the original resource Id of the database - though, CreateMode.Restore requires RestorableDroppedDatabaseId.
+            */
+            builder.Arguments.CreateMode = CreateMode.Restore;
+            builder.Arguments.RestorableDroppedDatabaseId = databaseId;
+            builder.Arguments.SourceDatabaseDeletionDate = deletionDate;
+            builder.Arguments.RestorePointInTime = pointInTime ?? null;
+            return builder;
+        }
+
+        /// <summary>
+        /// Sets this database to be a secondary replica of an existing database, using the existing primary database's resource ID.
+        /// </summary>
+        /// <param name="builder"></param>
+        /// <param name="databaseId">Resource ID of the primary database</param>
+        /// <returns></returns>
+        public static SqlDatabaseBuilder SetAsSecondary(this SqlDatabaseBuilder builder, Input<string> databaseId)
+        {
+            builder.Arguments.CreateMode = CreateMode.Secondary;
+            builder.Arguments.SourceDatabaseId = databaseId;
+            return builder;
+        }
+
+        /// <summary>
+        /// Sets the database to be a copy of an existing database
+        /// </summary>
+        /// <param name="builder"></param>
+        /// <param name="databaseId">Resource ID of the database to copy</param>
         /// <returns></returns>
-        public static SqlDatabaseBuilder RestoreFrom(this SqlDatabaseBuilder builder, Input<string> databaseId)
+        public static SqlDatabaseBuilder SetAsCopy(this SqlDatabaseBuilder builder, Input<string> databaseId)
         {
-            builder.Arguments.CreateMode = Pulumi.AzureNative.Sql.CreateMode.Restore;
+            builder.Arguments.CreateMode = CreateMode.Copy;
             builder.Arguments.SourceDatabaseId = databaseId;
             return builder;
         }
 
         /// <summary>
-        /// Enables the read scale
+        /// Sets the database to be a restoration of a geo-replicated backup database.
+        /// </summary>
+        /// <param name="builder"></param>
+        /// <param name="databaseId">Recoverable Database Resource ID of the database to restore</param>
+        /// <returns></returns>
+        public static SqlDatabaseBuilder SetAsRecovery(this SqlDatabaseBuilder builder, Input<string> databaseId)
```

Comment:
Should we rename this to "CreateAsRecoveryOf"? I feel the current name is a bit confusing.